src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,548 @@
+        __Module_init(proposal_, metadata);
+        //init empty list of bounties and claims
+        _bountyList.init();
+        _claimList.init();
+    }
+
+    //--------------------------------------------------------------------------
+    // Getter Functions
+
+    /// @inheritdoc IBountyManager
+    function getBountyInformation(uint bountyId)
+        external
+        view
+        validBountyId(bountyId)
+        returns (Bounty memory)
+    {
+        return _bountyRegistry[bountyId];
+    }
+
+    /// @inheritdoc IBountyManager
+    function listBountyIds() external view returns (uint[] memory) {
+        return _bountyList.listIds();
+    }
+
+    /// @inheritdoc IBountyManager
+    function isExistingBountyId(uint bountyId) public view returns (bool) {
+        return _bountyList.isExistingId(bountyId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function getClaimInformation(uint claimId)
+        external
+        view
+        validClaimId(claimId)
+        returns (Claim memory)
+    {
+        return _claimRegistry[claimId];
+    }
+
+    /// @inheritdoc IBountyManager
+    function listClaimIds() external view returns (uint[] memory) {
+        return _claimList.listIds();
+    }
+
+    /// @inheritdoc IBountyManager
+    function isExistingClaimId(uint claimId) public view returns (bool) {
+        return _claimList.isExistingId(claimId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function listClaimIdsForContributorAddress(address contributorAddrs)
+        external
+        view
+        returns (uint[] memory)
+    {
+        return contributorAddressToClaimIds[contributorAddrs].values();
+    }
+
+    //--------------------------------------------------------------------------
+    // Mutating Functions
+
+    /// @inheritdoc IBountyManager
+    function addBounty(
+        uint minimumPayoutAmount,
+        uint maximumPayoutAmount,
+        bytes calldata details
+    )
+        external
+        onlyRole(uint8(Roles.BountyAdmin))
+        validPayoutAmounts(minimumPayoutAmount, maximumPayoutAmount)
+        returns (uint id)
+    {
+        // Note ids start at 1.
+        uint bountyId = ++_nextId;
+
+        // Add Bounty id to the list.
+        _bountyList.addId(bountyId);
+
+        Bounty storage b = _bountyRegistry[bountyId];
+
+        b.minimumPayoutAmount = minimumPayoutAmount;
+        b.maximumPayoutAmount = maximumPayoutAmount;
+        b.details = details;
+
+        emit BountyAdded(
+            bountyId, minimumPayoutAmount, maximumPayoutAmount, details
+        );
+
+        return bountyId;
+    }
+
+    /// @inheritdoc IBountyManager
+    function updateBounty(uint bountyId, bytes calldata details)
+        external
+        onlyRole(uint8(Roles.BountyAdmin))
+        validBountyId(bountyId)
+    {
+        _bountyRegistry[bountyId].details = details;
+
+        emit BountyUpdated(bountyId, details);
+    }
+
+    /// @inheritdoc IBountyManager
+    function lockBounty(uint bountyId)
```

Comment:
Well, there is a `lockBounty` function, and I do not specifically know why it is here, but would it make sense to include another access controlled `unlockBounty` function?

Reply:
I would say no because if you wanted to change the Bounty in the first place you could have just used updateBounty
Even if lockBounty is final, you could just create a new Bounty
The initial intend for the function was to be able to actually clean up remaining bounties that werent reasonable to be fulfilled in the first place